var/spack/repos/builtin/packages/fenics2019/package.py

```diff
@@ -0,0 +1,144 @@
+
+    depends_on('py-pybind11', type=('build', 'run'))
+    depends_on('cmake@3.17.3:', type='build')
+
+    depends_on('py-setuptools', type='build', when='+python')
+    depends_on('py-pkgconfig', type='run', when='+python')
+    depends_on('py-sphinx@1.0.1:', when='+doc', type='build')
+
+    def cmake_is_on(self, option):
+        return 'ON' if option in self.spec else 'OFF'
+
+    def cmake_args(self):
+        spec = self.spec
+
+        opts = [
+            '-DDOLFIN_ENABLE_DOCS:BOOL={0}'.format(self.cmake_is_on('+doc')),
+            '-DBUILD_SHARED_LIBS:BOOL={0}'.format(self.cmake_is_on('+shared')),
+            '-DDOLFIN_SKIP_BUILD_TESTS:BOOL=ON',
+            '-DDOLFIN_ENABLE_OPENMP:BOOL={0}'.format(self.cmake_is_on('+openmp')),
+            '-DDOLFIN_ENABLE_CHOLMOD:BOOL={0}'.format(
+                self.cmake_is_on('suite-sparse')),
+            '-DDOLFIN_ENABLE_HDF5:BOOL={0}'.format(self.cmake_is_on('hdf5')),
+            '-DDOLFIN_ENABLE_MPI:BOOL={0}'.format(self.cmake_is_on('mpi')),
+            '-DDOLFIN_ENABLE_PARMETIS:BOOL={0}'.format(
+                self.cmake_is_on('parmetis')),
+            '-DDOLFIN_ENABLE_PASTIX:BOOL={0}'.format(self.cmake_is_on('pastix')),
+            '-DDOLFIN_ENABLE_PETSC:BOOL={0}'.format(self.cmake_is_on('petsc')),
+            '-DDOLFIN_ENABLE_PETSC4PY:BOOL={0}'.format(
+                self.cmake_is_on('py-petsc4py')),
+            '-DDOLFIN_ENABLE_PYTHON:BOOL={0}'.format(self.cmake_is_on('python')),
+            '-DDOLFIN_ENABLE_QT:BOOL={0}'.format(self.cmake_is_on('qt')),
+            '-DDOLFIN_ENABLE_SCOTCH:BOOL={0}'.format(self.cmake_is_on('scotch')),
+            '-DDOLFIN_ENABLE_SLEPC:BOOL={0}'.format(self.cmake_is_on('slepc')),
+            '-DDOLFIN_ENABLE_SLEPC4PY:BOOL={0}'.format(
+                self.cmake_is_on('py-slepc4py')),
+            '-DDOLFIN_ENABLE_SPHINX:BOOL={0}'.format(self.cmake_is_on('py-sphinx')),
+            '-DDOLFIN_ENABLE_TRILINOS:BOOL={0}'.format(
+                self.cmake_is_on('trilinos')),
+            '-DDOLFIN_ENABLE_UMFPACK:BOOL={0}'.format(
+                self.cmake_is_on('suite-sparse')),
+            '-DDOLFIN_ENABLE_VTK:BOOL={0}'.format(self.cmake_is_on('vtk')),
+            '-DDOLFIN_ENABLE_ZLIB:BOOL={0}'.format(self.cmake_is_on('zlib')),
+        ]
+        os.environ['PETSC_DIR'] = spec['petsc'].prefix
+        os.environ['SLEPC_DIR'] = spec['slepc'].prefix
+
+        return opts
+
+    # set environment for bulding python interface
+    def setup_build_environment(self, env):
+        env.set('DOLFIN_DIR', self.prefix)
+
+    def setup_run_environment(self, env):
+        env.set('DOLFIN_DIR', self.prefix)
+
+    # build python interface of dolfin
+    @run_after('install')
+    def install_python_interface(self):
+        if '+python' in self.spec:
+            cd('python')
+            python('setup.py', 'install',
+                   '--single-version-externally-managed',
+                   '--root=/', '--prefix={0}'.format(self.prefix))
```

Comment:
I don't know much about python, but just to be sure: what does `--root=/` do?

Reply:
The `--root=/` option is required by the `--single-version-externally-managed` flag and installs everything to the alternate root directory as specified. I found this option in the package from [JaroslavHron](https://github.com/JaroslavHron/spack_repo/tree/master/fenics/packages/fenics), and it provides a working way to get the dolfin python interface installed. Note that by only using the `--prefix` I did not manage to successfully import the dolfin package in python.

However I suggest to replace it with a simple `--user` flag, where the python packages are installed in the users site-package path and which I have successfully tested on my machine as well. 
`python('setup.py', 'install', '--user')`

Reply:
Pretty sure @adamjstewart has a more informed opinion about this

Reply:
The current logic is correct, we want to install to the package installation prefix, not a shared site-packages path. The only change I would suggest is to replace `python('setup.py', 'install'` with `setup_py('install'`, which adds additional logic to ignore user configs, making more reproducible builds.

Reply:
Suggest changes are reflected in latest commit.